tests/search/search-test.spec.ts

```diff
@@ -0,0 +1,210 @@
+  })
+
+  test('sort-filter-bar', async () => {
+    await page.waitForTimeout(5000);
+    expect(page.url()).toBe(queryURL);
+
+    const infiniteScroller = page.locator('infinite-scroller');
+    const sortFilterBar = page.locator('sort-filter-bar');
+    const sortContainer = sortFilterBar.locator('#container');
+    const sortBarSection = sortFilterBar.locator('section#sort-bar');
+
+    expect(await infiniteScroller.count()).toEqual(1);
+    expect(await sortFilterBar.count()).toEqual(1);
+    expect(await sortContainer.count()).toEqual(1);
+    expect(await sortBarSection.count()).toEqual(1);
+
+    const sortDirectionContainer = sortBarSection.locator('.sort-direction-container');
+    const sortByText = sortBarSection.locator('.sort-by-text');
+    const sortDirectionSelector = sortDirectionContainer.locator('.sort-direction-selector');
+    const srOnlySortDirection = sortDirectionContainer.locator('.sr-only');
+
+    expect(await sortDirectionContainer.count()).toEqual(1);
+    expect(await sortByText.count()).toEqual(1);
+    expect(await sortDirectionSelector.count()).toEqual(1);
+    expect(await srOnlySortDirection.count()).toEqual(1);
+
+    expect(await sortByText.innerText()).toEqual('Sort by:');
+    expect(await srOnlySortDirection.innerText()).toEqual('Change to ascending sort');
+
+    const sortSelectorContainer = sortBarSection.locator('div#sort-selector-container');
+    const mobileSort = sortSelectorContainer.locator('#mobile-sort-container');
+    const desktopSort = sortSelectorContainer.locator('#desktop-sort-container');
+    
+    const desktopSortSelector = desktopSort.locator('#desktop-sort-selector');
+    const desktopSortSelectorTexts = await desktopSort.locator('ul#desktop-sort-selector').allInnerTexts();
+    const desktopSortTexts = Object.assign([], desktopSortSelectorTexts[0].split('\n'));
+
+    await desktopSortSelector.click();
+    await expect(page).toHaveURL(/sort=title/);
+
+    await page.waitForTimeout(5000);
+    expect(await srOnlySortDirection.innerText()).toEqual('Change to descending sort');
+
+    desktopSortTexts.forEach((text: string, ix) => {
+      expect(text.includes(sortTextList[ix]));
+    });
+    
+    expect(await sortSelectorContainer.count()).toEqual(1);
+    expect(await desktopSortSelector.count()).toEqual(1);
+
+    expect(await expect(mobileSort).toHaveClass(/hidden/));
+    expect(await expect(desktopSort).toHaveClass(/visible/));
+
+    // sort-bar-filter URL changes
+    await page.getByRole('button', { name: 'Relevance' }).click();
+
+    await page.getByText('Weekly views').first().click();
+    await expect(page).toHaveURL(/query=cats&sort=-week/);
+
+    await page.getByRole('button', { name: 'Toggle options Weekly views' }).getByRole('button').click();
+    await page.getByRole('button', { name: 'All-time views' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=-downloads/);
+
+    await page.getByRole('button', { name: 'Title' }).click();
```

Comment:
I think it would be good, with each click - that it checks the results set being displayed...